# PAD - Praca domowa 3
## Jan Solarz s26342

### Zadanie 1

Zaczynamy od wgrania ramki danych

In [113]:
import pandas as pd
PATH = 'Downloads/bank.csv'

In [114]:
PATH

'Downloads/bank.csv'

In [115]:
df=pd.read_csv(PATH,sep=";")


In [116]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4521 entries, 0 to 4520
Data columns (total 12 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   age        4521 non-null   int64 
 1   job        4521 non-null   object
 2   marital    4521 non-null   object
 3   education  4521 non-null   object
 4   default    4521 non-null   object
 5   balance    4521 non-null   int64 
 6   housing    4521 non-null   object
 7   loan       4521 non-null   object
 8   contact    4521 non-null   object
 9   day        4521 non-null   int64 
 10  month      4521 non-null   object
 11  duration   4521 non-null   int64 
dtypes: int64(4), object(8)
memory usage: 424.0+ KB


In [117]:
df

age            job  marital  education default  balance housing loan  \
0      30     unemployed  married    primary      no     1787      no   no   
1      33       services  married  secondary      no     4789     yes  yes   
2      35     management   single   tertiary      no     1350     yes   no   
3      30     management  married   tertiary      no     1476     yes  yes   
4      59    blue-collar  married  secondary      no        0     yes   no   
...   ...            ...      ...        ...     ...      ...     ...  ...   
4516   33       services  married  secondary      no     -333     yes   no   
4517   57  self-employed  married   tertiary     yes    -3313     yes  yes   
4518   57     technician  married  secondary      no      295      no   no   
4519   28    blue-collar  married  secondary      no     1137      no   no   
4520   44   entrepreneur   single   tertiary      no     1136     yes  yes   

       contact  day month  duration  
0     cellular   19   oct        79  
1     cellular   11   may       220  
2     cellular   16   apr       185  
3      unknown    3   jun       199  
4      unknown    5   may       226  
...        ...  ...   ...       ...  
4516  cellular   30   jul       329  
4517   unknown    9   may       153  
4518  cellular   19   aug       151  
4519  cellular    6   feb       129  
4520  cellular    3   apr       345  

[4521 rows x 12 columns]

In [118]:
o = df['month'].value_counts()
o
#may 1398, dec 20

may    1398
jul     706
aug     633
jun     531
nov     389
apr     293
feb     222
jan     148
oct      80
sep      52
mar      49
dec      20
Name: month, dtype: int64

In [119]:
print("Największa ilość połączeń była w miesiącu", o.idxmax(axis = 1),"i wynosiła",o.max(),"połączeń.",
     "Odpowiednio najmniej mamy dla",  o.idxmin(axis = 1),"gdzie wynosi",o.min(),"połączeń.")

Największa ilość połączeń była w miesiącu may i wynosiła 1398 połączeń. Odpowiednio najmniej mamy dla dec gdzie wynosi 20 połączeń.


### Zadanie 2

In [120]:
o=round(df.groupby('month')['duration'].mean(),3)

o

month
apr    294.355
aug    243.742
dec    416.900
feb    255.698
jan    269.541
jul    271.205
jun    252.426
mar    198.898
may    267.316
nov    272.067
oct    272.800
sep    215.731
Name: duration, dtype: float64

In [121]:
print("Największy średni czas spędzony na rozmowach był w miesiącu:", o.idxmax(axis = 1),"i wynosił",o.max(),"minut.")

Największy średni czas spędzony na rozmowach był w miesiącu: dec i wynosił 416.9 minut.


### Zadanie 3

In [122]:
o=round(df.groupby('education')['balance'].mean(),3)
o
#tertiary     1775.424, secondary     1196.814

education
primary      1411.544
secondary    1196.814
tertiary     1775.424
unknown      1701.246
Name: balance, dtype: float64

Największe oszczędności posiadają ludzie o wykształceniu "tertiary": 1775, najmeniejsze grupa "secondary": 1196. Klienci o nieznanym wykształceniu mają oszczedności o średniej 1700. Nie można jednoznacznie stwierdzić że "Im wyższe wykształcenie tym lepsze oszczędności".

### Zadanie 4

In [123]:
b=df.groupby('marital')['age'].max()
b

marital
divorced    84
married     87
single      69
Name: age, dtype: int64

In [124]:
print("Najstarsza osoba w związku małżeńskim ma lat:",df[df['marital']=='married']['age'].max(), "natomiast najstarszy singiel ma lat:",df[df['marital']=='single']['age'].max())

Najstarsza osoba w związku małżeńskim ma lat: 87 natomiast najstarszy singiel ma lat: 69
